core/src/main/scala/org/apache/spark/deploy/master/WorkerInfo.scala

```diff
@@ -89,17 +122,52 @@ private[spark] class WorkerInfo(
     drivers(driver.id) = driver
     memoryUsed += driver.desc.mem
     coresUsed += driver.desc.cores
+    driverToResourcesUsed(driver.id) = driver.resources
   }
 
   def removeDriver(driver: DriverInfo) {
     drivers -= driver.id
     memoryUsed -= driver.desc.mem
     coresUsed -= driver.desc.cores
+    driverToResourcesUsed.remove(driver.id)
   }
 
   def setState(state: WorkerState.Value): Unit = {
     this.state = state
   }
 
   def isAlive(): Boolean = this.state == WorkerState.ALIVE
+
+  /**
+   * acquire specified amount resources for driver/executor from the worker
+   * @param resourceReqs the resources requirement from driver/executor
+   * @return
+   */
+  def acquireResources(resourceReqs: Map[String, Int])
+  : Map[String, Seq[String]] = {
+    resourceReqs.map { case (rName, amount) =>
+      // TODO (wuyi) rName does not exists ?
```

Comment:
ditto for `(wuyi)`.

Reply:
what does the comment mean?  Why wouldn't rName exist?